x/tokenfactory/keeper/createdenom_test.go

```diff
@@ -50,11 +48,70 @@ func (suite *KeeperTestSuite) TestMsgCreateDenom() {
 	suite.Require().Len(queryRes2.Denoms, 2)
 
 	// Make sure that a second account can create a denom with the same subdenom
-	res, err = msgServer.CreateDenom(sdk.WrapSDKContext(suite.Ctx), types.NewMsgCreateDenom(suite.TestAccs[1].String(), "bitcoin"))
+	res, err = suite.msgServer.CreateDenom(sdk.WrapSDKContext(suite.Ctx), types.NewMsgCreateDenom(suite.TestAccs[1].String(), "bitcoin"))
 	suite.Require().NoError(err)
 	suite.Require().NotEmpty(res.GetNewTokenDenom())
 
 	// Make sure that an address with a "/" in it can't create denoms
-	res, err = msgServer.CreateDenom(sdk.WrapSDKContext(suite.Ctx), types.NewMsgCreateDenom("osmosis.eth/creator", "bitcoin"))
+	res, err = suite.msgServer.CreateDenom(sdk.WrapSDKContext(suite.Ctx), types.NewMsgCreateDenom("osmosis.eth/creator", "bitcoin"))
 	suite.Require().Error(err)
 }
+
+func (suite *KeeperTestSuite) TestCreateDenom() {
+	for _, tc := range []struct {
+		desc     string
+		malleate func()
+		subdenom string
+		valid    bool
+	}{
+		{
+			desc:     "subdenom too long",
+			malleate: func() {},
```

Comment:
Lets delete the function arg here, and just do it in the main body. (e.g. if `tc.setup != nil { tc.setup() }`)